client/policy_validator.go

```diff
@@ -171,7 +168,13 @@ func (p *ClusterPolicyValidator) HasPermission() bool {
 }
 
 func (p *ClusterPolicyValidator) CrdDefined() bool {
+	if p.mockEnabled {
+		return true
+	}
 	if p.dc == nil {
+		if p.cli.RestConfig == nil {
+			return false
```

Comment:
Is this the right result? Feels like it is safer to fail the attempted operation if we cannot determine whether or not policy is enabled?

Reply:
The `RestConfig` attribute is always set, except when using `newMockClient` to create the `*VanClient` instance.
When it happens (mock tests only), we have to consider CRD as absent, so existing unit tests continue to work.

That's why we now have the `disablePolicyDiscovery` flag. If a unit test needs the policy validator to consider
CRD as present, we set it to true.